### PR TITLE
Throw error for improper node js version using http2

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

This feature spawned from a short discussion in #1335. Will be adding an appropriate unit test to this PR and documentation changes. It will slightly increase the overall code coverage as well.

### CODE REVIEW

**Comment by Ethan-Arrowood** [comment]
@cemremengu I just pushed a unit test and doc change. please review, thank you!

File: docs/HTTP2.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 _Fastify_ offers **experimental support** for HTTP2 starting from Node
 8.8.0, which includes HTTP2 without a flag. _Fastify_ supports HTTP2
-both over HTTPS or over plaintext.
+both over HTTPS or over plaintext. _Fastify_ will throw an `http2 is available only from node >= 8.8.1` error if HTTP2 is enabled on an invalid version of `Node.js`.
```
Comment (cemremengu) [inline docs/HTTP2.md]: I think you can shorten/emphasize this like: 
**Note that http2 is available only for node versions >= 8.8.1**

File: test/http2/missing-http2-module.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+'use strict'
+
+const t = require('tap')
+const test = t.test
+const proxyquire = require('proxyquire')
+const Fastify = proxyquire('../..', { http2: null })
+
+test('fastify throws error when http2 module cannot be found', t => {
```
Comment (cemremengu) [inline test/http2/missing-http2-module.js]: should throw when http2 module cannot be found

File: fastify.js [hunk 3]
```diff
@@ -989,7 +989,7 @@ function http2 () {
   try {
     return require('http2')
   } catch (err) {
-    console.error('http2 is available only from node >= 8.8.1')
+    throw new Error('http2 is available only from node >= 8.8.1')
```
Comment (mcollina) [inline fastify.js]: this should likely get an error code, see lib/errors.js

**Comment by mcollina** [comment]
Good work!

**Pull request merged**